src/gpu/optixPrograms.cu

```diff
@@ -154,27 +124,11 @@ extern "C" __global__ void __closesthit__()
 	const Vector<3, double> hwrd = hitWorldRaytraced;
 	const Vector<3, double> hso = beamSampleOrigin;
 	const double distance = (hwrd - hso).length();
```

Comment:
Not mine. However I may only guess this is because of some precision issues when calculating distance. Especially when you look at two vectors above, that are explicitly made double too. I would rather leave it, but I can also change it - what do you think? I casted distance to float, leaving double for length calculation and it seems to work - tests pass.

Reply:
Discussed offline. It was a precision issue at some point in the past. Does not seem to happen now, however going back to float would require some tests. I put a TODO here for now.